src/misc/ListModel.ts

```diff
@@ -250,19 +252,19 @@ export class ListModel<ElementType extends ListElement> {
 					this.rawState.filteredItems.length > 1 &&
 					!this.rawState.inMultiselect
 				) {
-					const nextSelection =
+					activeElement =
 						entity === last(this.state.items)
 							? this.state.items[this.state.items.length - 2]
 							: this.state.items[this.state.items.indexOf(entity) + 1]
 
-					selectedItems.add(nextSelection)
+					selectedItems.add(activeElement)
 				}
 
 				const filteredItems = this.rawState.filteredItems.slice()
 				remove(filteredItems, entity)
 				const unfilteredItems = this.rawState.unfilteredItems.slice()
 				remove(unfilteredItems, entity)
-				this.updateState({ filteredItems, selectedItems, unfilteredItems })
+				this.updateState({ filteredItems, selectedItems, unfilteredItems, activeElement: activeElement ?? entity })
```

Comment:
why should `entity` be an `activeElement` if it's deleted? shouldn't we keep the previous selection if it's not the selected item that got removed?